R/polars_options.R

```diff
@@ -0,0 +1,105 @@
+#' Get and reset polars options
+#'
+#' @description
+#' `polars_options()` returns a list of options for polars. Options
+#' can be set with [`options()`]. Note that **options must be prefixed with
+#' "polars."**, e.g to modify the option `conversion_int64` you need to pass
+#' `options(polars.conversion_int64 =)`. See below for a description of all
+#' options.
+#'
+#' `polars_options_reset()` brings all polars options back to their default
+#' value.
+#'
+#' @details The following options are available (in alphabetical order, with the
+#'   default value in parenthesis):
+#'
+#' * for all `conversion_*` options, see arguments of [to_r_vector()][series__to_r_vector].
+#' * `df_knitr_print` (TODO: possible values??)
+#'
+#' @return
+#' `polars_options()` returns a named list where the names are option names and
+#' values are option values.
+#'
+#' `polars_options_reset()` doesn't return anything.
+#'
+#' @export
+#' @examplesIf requireNamespace("withr", quietly = TRUE) && requireNamespace("hms", quietly = TRUE)
+#' library(hms)
+#' polars_options()
+#' withr::with_options(
+#'   list(polars.conversion_int64 = "character"),
+#'   polars_options()
+#' )
+polars_options <- function() {
+  out <- list(
+    df_knitr_print = getOption("polars.df_knitr_print") %||% "auto",
+    conversion_int64 = getOption("polars.conversion_int64") %||% "double",
+    conversion_uint8 = getOption("polars.conversion_uint8") %||% "integer",
+    conversion_date = getOption("polars.conversion_date") %||% "Date",
+    conversion_time = getOption("polars.conversion_time") %||% "hms",
+    conversion_decimal = getOption("polars.conversion_decimal") %||% "double",
+    conversion_ambiguous = getOption("polars.conversion_ambiguous") %||% "raise",
+    conversion_non_existent = getOption("polars.conversion_non_existent") %||% "raise"
```

Comment:
I don't think the word "conversion" should be used because it is not clear what is meant by conversion from something to something else.

Reply:
I put `to_r_vector`, e.g. `polars.to_r_vector_int64` but I'm also open to `polars.to_r_int64` to have shorter names.